### ISSUE TITLE
Handle file firewall forbidden message and show it to the user.

### ISSUE THREAD

**Issue by SergioBertolinSG** [owncloud/ios#494]
Currently iOS client is not showing the user what happened if the server blocks access by firewall.

To block the iOS client add a rule in the file firewall of the server (enterprise only) like this:

![screen shot 2015-10-13 at 17 21 48](https://cloud.githubusercontent.com/assets/9048796/10459230/ee33e88c-71ce-11e5-9d58-e76707d85c9e.png)

For example the desktop client blocked by firewall:

<img src="https://cloud.githubusercontent.com/assets/9048796/10459377/87efff92-71cf-11e5-9e9b-a4506db8ed7e.png" width="420">

cc: @nasli @rperezb 

- [x] [GIT] Create branch manage_new_error_messages on the main app
- [x] [GIT] Create branch manage_new_error_messages_library on the library
- [x] [DEV] Modify the XML parser to detect the "Message field"
- [x] [DEV] Refactor the code to manage all the errors on all the request with the same method
- [x] [DEV] Return the error message inside the NSError for special cases like Firewall
- [x] [DEV] Use the new Error code OCErrorForbidenUnknow to show the right error message on the uploads view
- [x] [DEV] Control on the ManageErrros class the OCErrorForbidenUnknow to show the right error on all the cases
- [x] [DEV] Use the string from the server instead a translated message
- [x] [DEV] Show the right error on the Login View
- [x] [DEV] Show the right error when we make a download
- [x] [DEV] Show the right error when the user try to upload a file
- [x] [DEV] Show the right error on the Document Provider
- [x] [DEV] Show the right error on the Share In
- [X] [BUG]  Show different error message when firewall blocks than when user doesn't have write permission in the folder ( https://github.com/owncloud/ios/issues/494#issuecomment-297983792 ) [FIXED]
- [X] Not clear message when downloading with a firewall rule   [FIXED]
- [X] [QA] Create test plan
- [X] [QA] Validate test plan @jesmrec
- [ ] [GIT] Merge branch manage_new_error_messages on the main app
- [ ] [GIT] Merge branch manage_new_error_messages_library on the library

PR:https://github.com/owncloud/ios/pull/851
PR library: https://github.com/owncloud/ios-library/pull/148

**Comment by nasli** [comment]
The server is using the same code status for many different errors. The desktop client is showing the message that receive from the server, but what happen if you have other language, the error message shown in another language... not cool, we have to think to use different code status.

**Comment by SergioBertolinSG** [comment]
cc @nickvergessen

**Comment by javiergonzper** [comment]
In our login view we should show the "firewall error" in the same place where we show a login error.
Also maybe we could receive the error after the login because the admin can activate the firewall after the user log in.

**Comment by nickvergessen** [comment]
> The desktop client is showing the message that receive from the server, but what happen if you have other language.

The server response in the language that the user selected.

The status is `403 - Forbidden` is the only status code that matches.

**Comment by jesmrec** [comment]
Needed a review of the status flow for the banned uploads. Right now we have different behaviours depending of the rule/issue that bans the upload. 

CC @nasli

**Comment by nasli** [comment]
Agree @jesmrec. 

Summarizing, after an error is detected, the upload should fail and have the chance to retry.

**Comment by davivel** [comment]
Fixing last loose end...

**Comment by davivel** [comment]
Fixed messages, @jesmrec  please test.

**Comment by jesmrec** [comment]
@davivel, the behaviour right now is setting the same message in both banned uploads due to firewall rule and lack of permissions. `Upload not allowed`.  This is the same behaviour as before but with a different message.

**Comment by davivel** [comment]
Too easy to be true; checking.

**Comment by davivel** [comment]
Now it should be fixed; @jesmrec , please, test again.

**Comment by jesmrec** [comment]
Setting some rules like:

- Client device is ...
- User groups is ...
- Request time between ...

and trying to download a file, the error message is 'Cannot establish connection'. This message does not show what is really happening. Could it be improved?

@davivel @nasli

**Comment by rperezb** [comment]
fyi there is a blocking issue https://github.com/owncloud/firewall/issues/266

**Comment by davivel** [comment]
Updated message for downloads.

**Comment by jesmrec** [comment]
Bug fixed. Approved, but pending the server issue to be fixed.

As tech debt, the uploads view should be refactored to set retries in case they are needed (like in the case of lack of quota)

CC @nasli @davivel

**Comment by davivel** [comment]
@jesmrec , could you open a new issue for the retries?

Thanks
